### ISSUE TITLE
zval---Setting the value  example code

### ISSUE THREAD

**Issue by huqinlou0123** [phpinternalsbook/PHP-Internals-Book#38]
[link](https://github.com/phpinternalsbook/PHP-Internals-Book/blob/master/Book/php7/internal_types/zvals/basic_structure.rst#setting-the-value)

```
PHP_FUNCTION(hello_world) {
    Z_TYPE_P(return_value) = IS_STRING;
    Z_STR_P(return_value) = zend_string_init("hello world!", strlen(""hello world!"), 0);
};

/* ... */
    PHP_FE(hello_world, NULL)
/* ... */
```

Above   Z_TYPE_P(return_value) = IS_STRING;

#define Z_TYPE_P(zval_p)			Z_TYPE(*(zval_p))
#define Z_TYPE(zval)				zval_get_type(&(zval))	//获取zval类型
static zend_always_inline zend_uchar zval_get_type(const zval* pz) {
	return pz->u1.v.type;
}

this way is it wrong?

**Comment by nikic** [comment]
Yeah, this doesn't work. This was taken over from the PHP 5 version, where this used to work. In PHP 7 one should use `ZVAL_STR` or `ZVAL_STRING`.

**Comment by huqinlou0123** [comment]
Z_TYPE_P  is use to read , not set?
